src/peergos/server/storage/IpfsWrapper.java

```diff
@@ -207,7 +208,16 @@ public static IpfsConfigParams buildConfig(Args args) {
 
     private final BlockMetadataStore metaDB;
 
-    public IpfsWrapper(Path ipfsDir, IpfsConfigParams ipfsConfigParams, BlockMetadataStore metaDB) {
+    private final BlockRequestAuthoriser blockRequestAuthoriser;
+
+    private final Supplier<Connection> dbConnectionPool;
```

Comment:
Aren't the batcave and connpool implementation details of the authoriser? Why do they need to be member fields?

Reply:
I see why they're there now, but it feels weird and tightly coupled.